scanners/nodejs/nodejs.go

```diff
@@ -0,0 +1,183 @@
+
+	packageManagerInputTitle   = "Package Manager"
+	packageManagerInputSummary = "The package manager used in the project"
+	packageManagerInputEnvKey  = "NODEJS_PACKAGE_MANAGER"
+
+	nodeVersionInputTitle   = "Node Version"
+	nodeVersionInputSummary = "The version of Node.js used in the project. To use the latest Node version, leave this empty"
+	nodeVersionInputEnvKey  = "NODEJS_VERSION"
+
+	lintWorkflowID = models.WorkflowID("lint")
+	testWorkflowID = models.WorkflowID("test")
+)
+
+type packageManager struct {
+	name     string
+	lockFile string
+}
+
+var pkgManagers = []packageManager{
+	{"npm", "package-lock.json"},
+	{"yarn", "yarn.lock"},
+	// Potential future package managers
+	//{"pnpm", "pnpm-lock.yaml"},
+	//{"bun", "bun.lock"},
+}
+
+// Scanner implements the Scanner interface for Node.js projects
+type Scanner struct {
+	searchDir       string
+	packageJsonPath string
+	packageManager  string
+	nodeVersion     string
+	buildScript     string
+	lintScript      string
+	testScript      string
+	scripts         []string
+}
+
+// NewScanner creates a new scanner instance.
+func NewScanner() *Scanner {
+	return &Scanner{}
+}
+
+// Name returns the name of the scanner
+func (scanner *Scanner) Name() string {
+	return ScannerName
+}
+
+// DetectPlatform checks if the given search directory contains a Node.js project
+func (scanner *Scanner) DetectPlatform(searchDir string) (bool, error) {
+	scanner.packageJsonPath = checkPackageJSON(searchDir)
+	if scanner.packageJsonPath == "" {
+		log.TPrintf("Platform not detected")
+		return false, nil
+	}
+
+	scanner.nodeVersion = checkNodeVersion(searchDir)
+	scanner.packageManager = checkPackageManager(searchDir)
+	scanner.scripts = checkPackageScripts(scanner.packageJsonPath)
+
+	if slices.Contains(scanner.scripts, "build") {
+		log.TPrintf("- build - found")
+		scanner.buildScript = "build"
+	} else {
+		log.TPrintf("- build - not found")
+	}
+
+	if slices.Contains(scanner.scripts, "lint") {
+		log.TPrintf("- lint - found")
+		scanner.lintScript = "lint"
+	} else {
+		log.TPrintf("- lint - not found")
+	}
+
+	if slices.Contains(scanner.scripts, "test") {
+		log.TPrintf("- test - found")
+		scanner.testScript = "test"
+	} else {
+		log.TPrintf("- test - not found")
+	}
+
+	log.TSuccessf("Platform detected")
+
+	return true, nil
+}
+
+func (scanner *Scanner) ExcludedScannerNames() []string {
+	return []string{}
+}
+
+// Options returns the options for the scanner
+func (scanner *Scanner) Options() (models.OptionNode, models.Warnings, models.Icons, error) {
+	projectRootOption := models.NewOption(projectRootDirInputTitle, projectRootDirInputSummary, projectRootDirInputEnvKey, models.TypeSelector)
+	nodeVersionOption := models.NewOption(nodeVersionInputTitle, nodeVersionInputSummary, nodeVersionInputEnvKey, models.TypeOptionalUserInput)
+	packageManagerOption := models.NewOption(packageManagerInputTitle, packageManagerInputSummary, packageManagerInputEnvKey, models.TypeSelector)
```

Comment:
We ask for the node version and package manager from the user. But the scanner tries to find these values from the project itself. Should not we only ask for these if the scanner could not determine it?